jupyterlab/commands.py

```diff
@@ -1208,17 +1212,25 @@ def _get_extensions_in_dir(self, dname, core_data):
             name = data['name']
             jlab = data.get('jupyterlab', dict())
             path = osp.abspath(target)
+
+            filename = osp.basename(target)
+            if filename.startswith("pin:"):
+                alias = filename[len("pin:"):-len(".tgz")]
```

Comment:
Will not this include the SHA into `alias`?

Reply:
Not in my tests. This filename is the package name plus the version.

Reply:
Ah, the SHA is only for local packages (`is_dir == True`). Feel free to mark as resolved if we think that will not happen (local install + pin).

Reply:
Ah, good point... I should test the local case.